clients/spark/src/main/scala/io/treeverse/clients/StorageUtils.scala

```diff
@@ -127,7 +132,7 @@ object StorageUtils {
 
     private def initializeS3Client(
         configuration: ClientConfiguration,
-        credentialsProvider: Option[AWSCredentialsProvider],
+        credentialsProvider: Option[Any],
```

Comment:
Is it possible to keep interface of the same type that `builderWithEndpoint.withCredentials` accepts instead of general object?